### PR TITLE
Update managed-identity.md

### PR DESCRIPTION
**Pull request by localden**

### CODE REVIEW

**Review by neha-bhargava — APPROVED** [review]
Few suggestions otherwise looks good. Thanks!

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 1]
```diff
@@ -1,51 +1,79 @@
...
+* [Azure Arc](/azure/azure-arc/overview)
+* [Azure Cloud Shell](/azure/cloud-shell/overview)
+* [Azure Service Fabric](/azure/service-fabric/service-fabric-overview)
 
-## How to use managed identity in MSAL
+For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
+
+## How to use managed identities
+
+There are two types of managed identities available to developers - **system-assigned** and **user-assigned**. You can learn more about the differences in the [Managed identity types](/azure/active-directory/managed-identities-azure-resources/overview#managed-identity-types) article. MSAL.NET supports acquiring tokens through both. Standard [MSAL.NET logging](/azure/active-directory/develop/msal-logging-dotnet) is possible to keep track of requests and related metadata.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:23]: nit?
```suggestion
There are two types of managed identities available to developers - **system-assigned** and **user-assigned**. You can learn more about the differences in the [Managed identity types](/azure/active-directory/managed-identities-azure-resources/overview#managed-identity-types) article. MSAL.NET supports acquiring tokens with both. With [MSAL.NET logging](/azure/active-directory/develop/msal-logging-dotnet) allows to keep track of requests and related metadata.
```

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 2]
```diff
@@ -1,51 +1,79 @@
...
+
+Prior to using managed identities from MSAL.NET, developers must enable them for the resources they want to use through Azure CLI or the Azure Portal.
+
+## Examples
+
+For both user-assigned and system-assigned identities, developers can use the <xref:Microsoft.Identity.Client.ManagedIdentityApplicationBuilder> class. Because the feature is experimental, using <xref:Microsoft.Identity.Client.BaseAbstractApplicationBuilder%601.WithExperimentalFeatures(System.Boolean)> is required.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:29]: Seems like there a few tricks to use, like these xref, when working with the docs. Unfortunately these don't render in GitHub PRs properly...

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 3]
```diff
@@ -1,51 +1,79 @@
...
+
+### System-assigned managed identities
+
+For system-assigned managed identities, the developer does not need to pass any additional information when creating an instance of <xref:Microsoft.Identity.Client.IManagedIdentityApplication>, as it will automatically infer the relevant metadata about the assigned identity.
+
+<xref:Microsoft.Identity.Client.IManagedIdentityApplication.AcquireTokenForManagedIdentity(System.String)> is called with the resource to acquire a token for, such as `https://management.azure.com`.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:35]: Add a note here? `Note that this resource format is different from the scopes format that are passed into other acquire token methods.`

**Review by pmaytak — APPROVED** [review]

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 4]
```diff
@@ -1,51 +1,79 @@
...
-* IMDS (VMs)
-* Azure Arc
+For failed requests the error response contains a correlation ID that can be used for further diagnostics and log analysis. Keep in mind that the correlation IDs generated in MSAL.NET or passed into MSAL are different than the one returned in server error responses, as MSAL.NET cannot pass the correlation ID to managed identity token acquisition endpoints.
 
-In Progress:
+## Potential errors
 
-* Cloud Shell
-* Service Fabric
+### `MsalServiceException` Error Code: `managed_identity_failed_response` Error Message: An unexpected error occurred while fetching the AAD token
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:73]: nit: I would make Potential errors a sub section of troubleshooting, and error sections smaller heading for easier reading.

File: msal-dotnet-articles/advanced/managed-identity.md (hunk 2 above)
Comment (localden) [reply msal-dotnet-articles/advanced/managed-identity.md:29]: Yeah, that's a trade-off with proper cross-doc references :/

**Review by neha-bhargava — APPROVED** [review]

**Pull request merged**